scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -796,3 +801,65 @@ def test_other_axis_tuples(axis):
     res2 = stats.mannwhitneyu(x2, y2, axis=1)
 
     np.testing.assert_array_equal(res, res2)
+
+
+def test_gmean_mixed_mask_nan_weights():
+    # targeted test of _axis_nan_policy_factory with 2D masked sample:
+    # omitting samples with masks and nan_policy='omit' are equivalent
+    # also checks paired-sample sentinel value removal
+    m, n = 3, 20
+    axis = -1
+
+    np.random.seed(0)
+    a = np.random.rand(m, n)
+    b = np.random.rand(m, n)
+    mask_a1 = np.random.rand(m, n) < 0.2
+    mask_a2 = np.random.rand(m, n) < 0.1
+    mask_b1 = np.random.rand(m, n) < 0.15
+    mask_b2 = np.random.rand(m, n) < 0.15
+    mask_a1[2, :] = True
+
+    a_nans = a.copy()
+    b_nans = b.copy()
+    a_nans[mask_a1 | mask_a2] = np.nan
+    b_nans[mask_b1 | mask_b2] = np.nan
+
+    a_masked1 = np.ma.masked_array(a, mask=mask_a1)
+    b_masked1 = np.ma.masked_array(b, mask=mask_b1)
+    a_masked1[mask_a2] = np.nan
+    b_masked1[mask_b2] = np.nan
+
+    a_masked2 = np.ma.masked_array(a, mask=mask_a2)
+    b_masked2 = np.ma.masked_array(b, mask=mask_b2)
+    a_masked2[mask_a1] = np.nan
+    b_masked2[mask_b1] = np.nan
+
+    a_masked3 = np.ma.masked_array(a, mask=(mask_a1 | mask_a2))
+    b_masked3 = np.ma.masked_array(b, mask=(mask_b1 | mask_b2))
+
+    mask_all = (mask_a1 | mask_a2 | mask_b1 | mask_b2)
+    a_masked4 = np.ma.masked_array(a, mask=mask_all)
+    b_masked4 = np.ma.masked_array(b, mask=mask_all)
+
+    res = stats.gmean(a_nans, weights=b_nans,
+                      nan_policy='omit', axis=axis)
+    res1 = stats.gmean(a_masked1, weights=b_masked1,
+                        nan_policy='omit', axis=axis)
```

Comment:
```suggestion
                       nan_policy='omit', axis=axis)
```
`scipy/stats/tests/test_axis_nan_policy.py:847:25: E127 continuation line over-indented for visual indent`